src/org/omegat/core/dictionaries/DictionaryEntry.java

```diff
@@ -31,14 +31,26 @@
  * @author Alex Buloichik (alex73mail@gmail.com)
  */
 public class DictionaryEntry implements Comparable<DictionaryEntry> {
+    private final String query;
     private final String word;
     private final String article;
 
     public DictionaryEntry(final String word, final String article) {
+        this.query = word.toLowerCase();
         this.word = word;
         this.article = article;
     }
 
+    public DictionaryEntry(final String query, final String word, final String article) {
```

Comment:
question: how to allow plugin developer to use this safely with compatible older omegat.

Reply:
ref https://wiki.eclipse.org/Evolving_Java-based_APIs

Reply:
It works file for plugins built with OmegaT 5.7. 
When dictionary plugin adopt new interface, the plugin will only work with newer OmegaT.